Initial Mapping Features

irinazoccolini opened the pull request:
This PR adds:
- initial map features such as a time slider and drop down menu
- hard-coded location markers on the map that are visible at different years (1913 & 1915)

JusticeV452 committed:
Revert "Updated map to be more dynamic"

This reverts commit dc8d51c648e5e3b70538bcfd8e6009817ee50d44.

JusticeV452 committed:
Updated map to be more dynamic

- Map uses testdata to render pointers
- Drew polygon on the map and made polygon clickable

JusticeV452 committed:
Updated variable names and code formatting

JusticeV452 committed:
Added material-ui time slider framework

JusticeV452 committed:
Added test dropdown component

JusticeV452 committed:
Merge branch 'main' into story-mapping

JusticeV452 committed:
Fixed missing Polygon import

irinazoccolini committed:
Move changes to MapMicro and add three address markers

irinazoccolini committed:
Fix component name

keileenorthcutt committed:
Merge branch 'story-mapping' of /Users/keileenorthcutt/Documents/GitHub/self-sustaining-cities with conflicts.

keileenorthcutt committed:
Merge remote-tracking branch 'origin/story-mapping' into story-mapping

# Conflicts:
#	webpack-stats.json

keileenorthcutt committed:
locations for 1913

JusticeV452 committed:
Change time slider to display set of locations with matching year

aponce215 committed:
Updated MapMicro Time Slider

User is now able to have inputs out of range. Slider UI will only update if input is within correct range

aponce215 committed:
Merge branch 'story-mapping' of https://github.com/dhmit/self-sustaining-cities into story-mapping

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
-    position: [38.8051606, -77.0036513]
-}];
+const ADDRESS_DATA = [
+    {
```
Great job keeping things simple as MVP, putting data directly into the component.

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
+//     [38.8051606, -77.0136513],
+//     [38.8051606, -77.0006513]
+// ];
+
```
Nitpick: please remove commented out code that's no longer useful!

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
+    sliderName, currentRange, defaultRange, lastValid, sliderChangeFunc, inputChangeFunc, sliderBlurFunc
+) {
+    const [minValue, maxValue] = defaultRange;
+    console.log("This is the last valid", lastValid);
```
Nitpick: remove console.log

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
 import React from "react";
 import {MapContainer, Marker, Popup, TileLayer, Polygon} from "react-leaflet";
```
remove unused import (Polygon)

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
+                        {/*    eventHandlers={{click: () => {console.log("marker clicked");}}}*/}
+                        {/*    pathOptions={{color: "purple"}}*/}
+                        {/*    positions={TEST_REGION}*/}
+                        {/*/>*/}
```
Nitpick: please remove unused code!

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
+            }
+            else{
+                newValidState = [newValue, currentUpperValue];
+            }
```
Is there a different way to write this function? Maybe:
```javascript 
If (newValue > minValue && < currentUpperValue) {
  newValidState = [newValue, currentUpperValue];
}
```
Please refactor if possible!

anastasia commented on frontend/components/MapMicro.js:
```diff
@@ -1,58 +1,319 @@
 
+    setSliderValue(newLowerBound, newUpperBound) {
+        this.setState({
+            ...this.state,
```
I don't think you need to pass in the state object every time to update state properties. Can remove.

anastasia reviewed (changes requested):
Hey, great work, team! I added the smallest nitpicks so please update that, and then this is ready to merge. I know there are some merge conflicts, but they are more complicated here so I will deal with that once your PR is ready.

aponce215 committed:
Refactored handleSliderInputChange

Refactored all the if statements in the bounds to be simpler and easier to understand

irinazoccolini committed:
Removed unused code

irinazoccolini committed:
Changed sliderState to lastValid

JusticeV452 committed:
Replaced multiselect-dropdown with react-bootstrap dropdown

- Test dropdown lists scrollable address data

irinazoccolini committed:
Removed unused import

anastasia committed:
fix - syntax errors, lint errors

anastasia committed:
Merge branch 'main' of github.com:dhmit/self-sustaining-cities into story-mapping

# Conflicts:
#	.idea/misc.xml
#	package-lock.json
#	webpack-stats.json

aponce215 committed:
Blanks in Slider Input

Slider inputs can now handle blanks, and whenever the client clicks outside of the input, the input's value is reset to the most recent valid input

aponce215 committed:
Merge branch 'story-mapping' of https://github.com/dhmit/self-sustaining-cities into story-mapping
